### DISCUSSION TITLE
Textarea trim option

### DISCUSSION THREAD

**Discussion by mehmetranas** [Trendyol/baklava#593] [General]
In the text area, we currently have a validation that requires a minimum length of 10 characters. However, we have noticed that if a user adds spaces before or after the text, the validation still passes even if the actual length is less than 10. To address this issue, we propose implementing a trim option that trims the text area value before performing the validation. This way, any leading or trailing spaces will be removed, ensuring accurate validation results.

Additionally, I attempted to solve the problem by utilizing the "bl-invalid" event, but unfortunately, it did not work as expected. Therefore, I am uncertain if the "bl-invalid" event can resolve the issue I am facing.

**Comment by muratcorlu** [comment]
Good question. I think we need some investigation on this to find out to best place to fix this issue. The same issue exists for native input and textarea elements as well. I think we can either add an option for minlength validation to trim the value before checking it or have a look giving an opportunity to the developer to make that kind of checks by themself. (like custom validation support for input elements)
